### PR TITLE
Support of PhaseTapChangerTapPositionAction for 3wt

### PR DESCRIPTION
**Pull request by vmouradian**
**Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

**What kind of change does this PR introduce?**
This PR enables to apply a pst action on a Three Windings Transformer

**What is the current behavior?**
Three Windings Transformers are not supported by pst actions

**What is the new behavior (if this is a feature change)?**
Three Windings Transformers are supported by pst actions

**Does this PR introduce a breaking change or deprecate an API?**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

### CODE REVIEW

**Review by Hadrien-Godard — APPROVED** [review]

File: src/main/java/com/powsybl/openloadflow/network/LfAction.java [hunk 1]
```diff
@@ -183,10 +180,22 @@ private static Optional<LfAction> create(LoadAction action, LfNetwork lfNetwork,
     }
 
     private static Optional<LfAction> create(PhaseTapChangerTapPositionAction action, LfNetwork lfNetwork) {
-        LfBranch branch = lfNetwork.getBranchById(action.getTransformerId()); // only two windings transformer for the moment.
+
+        String legNumber = "";
+
+        Optional<ThreeWindingsTransformer.Side> side = action.getSide();
+        if (side.isPresent()) {
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/network/LfAction.java:187]: You already have this need for `PropagatedContingency`, so we have to refactor to share code. Note that the initialization of leg number with `""` is not really nice. Better put something like:
````
 LfBranch branch = lfNetwork.getBranchById(side.isPresent() ? action.getTransformerId() + legNumber : action.getTransformerId());
````

File: src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java [hunk 2]
```diff
@@ -142,10 +142,13 @@ protected static void checkActions(Network network, List<Action> actions) {
                 case PhaseTapChangerTapPositionAction.NAME: {
                     PhaseTapChangerTapPositionAction phaseTapChangerTapPositionAction = (PhaseTapChangerTapPositionAction) action;
                     phaseTapChangerTapPositionAction.getSide().ifPresentOrElse(side -> {
-                        throw new PowsyblException("3 windings transformers not yet supported");
+
+                        if (network.getThreeWindingsTransformer(phaseTapChangerTapPositionAction.getTransformerId()) == null) {
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java:146]: same: could you try to refactor throws?

**Review by annetill — CHANGES_REQUESTED** [review]
Small clean needed, but functionally right.

File: src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java [hunk 3]
```diff
@@ -487,13 +485,12 @@ private static void createSwitches(List<Switch> switches, LfNetwork lfNetwork, L
         }
     }
 
-    private static void createPhaseControl(LfNetwork lfNetwork, PhaseTapChanger ptc, String controllerBranchId, String legId,
+    private static void createPhaseControl(LfNetwork lfNetwork, PhaseTapChanger ptc, String controllerBranchId,
                                            LfNetworkParameters parameters) {
         if (ptc != null && ptc.isRegulating() && ptc.getRegulationMode() != PhaseTapChanger.RegulationMode.FIXED_TAP) {
             String controlledBranchId = ptc.getRegulationTerminal().getConnectable().getId();
-            if (controlledBranchId.equals(controllerBranchId)) {
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java:494]: @geofjamg this part is really strange: no memory of such a that kind of code that I don't really get.

**Review by annetill — APPROVED** [review]

**Pull request merged**